pantheon/src/main/java/tech/pegasys/pantheon/cli/util/TomlConfigFileDefaultProvider.java

```diff
@@ -140,4 +144,22 @@ private void loadConfigurationFromFile() {
 
     checkConfigurationValidity();
   }
+
+  private void checkUnknownOptions(final TomlParseResult result) {
+    final CommandSpec commandSpec = commandLine.getCommandSpec();
+
+    final Set<String> unknownOptionsList =
+        result.keySet().stream()
+            .filter(option -> !commandSpec.optionsMap().containsKey("--" + option))
+            .collect(Collectors.toSet());
```

Comment:
Does this handle options with multiple names correctly?  e.g. `--rpc-http-api` can also be `--rpc-http-apis` - our actual lookup code does handle it so we should consider either version known.

Reply:
@ajsutton It handles such names correctly because the `commandSpec` is already initialised with all the valid options. I've updated the PantheonCommandTest to test the scenario as well.

Reply:
Awesome, LGTM then.